poetry/config/config.py

```diff
@@ -128,7 +129,11 @@ def process(self, value):  # type: (Any) -> Any
         if not isinstance(value, basestring):
             return value
 
-        return re.sub(r"{(.+?)}", lambda m: self.get(m.group(1)), value)
+        return re.sub(
+            r"{(.+?)}",
+            lambda m: self.get(m.group(1)) or "{{{key}}}".format(key=m.group(1)),
```

Comment:
This lambda is getting difficult to read at first pass. I'd suggest to take encapsulate this into its own function.

Reply:
I think I extracted it well enough [here](https://github.com/python-poetry/poetry/pull/3057/files#diff-302fd6e399fc926506961839561ed7635a346a4792af1d47d2cfa07c5651b7fbR138), should I change something? I'd still like to get this merged (and stop worrying about conflicts with upstream)